File: L1137-NthTribonacciNumber.java
// https://leetcode.com/problems/n-th-tribonacci-number
class Solution {
    public int tribonacci(int n) {
        int zero = 0,
            one = 1,
            two = 1,
            t1, t2;
        for (int i = 0; i < n; i++) {
            t1 = one;
            t2 = two;
            two = zero + one + two;
            zero = t1;
            one = t2;
        }
        return zero;
    }
}

File: L1137-nth_tribonacci_number.py
# https://leetcode.com/problems/n-th-tribonacci-number
class Solution:
    def tribonacci(self, n: int) -> int:
        zero, one, two = 0, 1, 1
        for i in range(n):
            t1 = one
            t2 = two
            two = zero + one + two
            zero = t1
            one = t2
        return zero
